requirements.txt

```diff
@@ -5,7 +5,7 @@ cookiecutter~=1.7.0
 dynaconf>=3.1.2,<4.0.0
 fsspec>=2021.4, <=2022.1
 gitpython~=3.0
-importlib_metadata>=4.4 # Compatible with Python 3.10 importlib.metadata
+importlib_metadata>=3.6  # The "selectable" entry points were introduced in `importlib_metadata` 3.6 and Python 3.10.
```

Comment:
I find this comment slightly confusing.. Do you mean selectable entry points only work with Python 3.10, or that `importlib_metadata` 3.6 introduces selectable entry points and is also compatible with python 3.10?

Reply:
`importlib_metadata` introduces new features that are later merged into CPython `importlib.metadata`. In this case, the "selectable" entry points feature--used by Kedro--was introduced in `importlib_metadata` 3.6, and thus also included in `importlib.metadata` natively in 3.10 (actually, `importlib_metadata` 4.4 was merged into Python 3.10, but Python 3.8 only has version `importlib_metadata` 1.4 features, and Python 3.9 didn't introduce anything on this front).

The comment is actually just copied from the compatibility note at the end of [this section](https://docs.python.org/3.10/library/importlib.metadata.html#entry-points).

Reply:
As mentioned in https://github.com/kedro-org/kedro/pull/1594#discussion_r892941129, I don't think it's worth doing a conditional import. We can drop the `importlib_metadata` third-party dependency when Kedro drops support for Python 3.9 lol.